src/main/java/seedu/address/logic/commands/SubscribeCommand.java

```diff
@@ -52,24 +55,30 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
             throw new CommandException(MESSAGE_DUPLICATE_FEED);
         }
 
-        EntryBook feedEntries;
+        URL feedUrl = toSubscribe.getLink().value;
+        SyndFeed feed;
         try {
             // we ensure the link is a feed here
-            feedEntries = FeedUtil.fromFeedUrl(toSubscribe.getLink().value);
+            feed = FeedUtil.fetchAsFeed(feedUrl);
         } catch (IOException e) {
             throw new CommandException(String.format(MESSAGE_FAILURE_NET, e));
         } catch (FeedException e) {
             throw new CommandException(String.format(MESSAGE_FAILURE_XML, toSubscribe.getLink().value));
         }
 
-        model.addFeedsEntry(toSubscribe);
+        EntryAutofill autofill = new EntryAutofill(toSubscribe);
+        autofill.extractFromFeed(feed);
```

Comment:
See below for related comment - maybe we should add n `autofill.extractFromFeedUrl(feedUrl)` before this,, in the unlikely case the feed does not have a title or description ([NYT](http://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml)).

Reply:
imo `autofill#extractFromUrl` does not really distinguish between whether the URL resource is an article or feed or something else, so maybe reuse it for now?